src/components/select/select.ts

```diff
@@ -0,0 +1,377 @@
+
+  @queryAssignedElements({ flatten: true, selector: 'igc-select-item' })
+  protected override items!: Array<IgcSelectItemComponent>;
+
+  @queryAssignedElements({ flatten: true, selector: 'igc-select-group' })
+  protected override groups!: Array<IgcSelectGroupComponent>;
+
+  @queryAssignedElements({ slot: 'helper-text' })
+  protected helperText!: Array<HTMLElement>;
+
+  @queryAssignedElements({ slot: 'suffix' })
+  protected inputSuffix!: Array<HTMLElement>;
+
+  @queryAssignedElements({ slot: 'prefix' })
+  protected inputPrefix!: Array<HTMLElement>;
+
+  @state()
+  protected override selectedItem!: IgcSelectItemComponent | null;
+
+  @query('igc-input')
+  protected override target!: IgcInputComponent;
+
+  /** The value attribute of the control. */
+  @property({ reflect: false, type: String })
+  public value!: string | undefined;
+
+  /** The name attribute of the control. */
+  @property()
+  public name!: string;
+
+  /** The disabled attribute of the control. */
+  @property({ reflect: true, type: Boolean })
+  public disabled = false;
+
+  /** The required attribute of the control. */
+  @property({ reflect: true, type: Boolean })
+  public required = false;
+
+  /** The invalid attribute of the control. */
+  @property({ reflect: true, type: Boolean })
+  public invalid = false;
+
+  /** The outlined attribute of the control. */
+  @property({ reflect: true, type: Boolean })
+  public outlined = false;
+
+  /** The autofocus attribute of the control. */
+  @property({ type: Boolean })
+  public override autofocus!: boolean;
+
+  /** The label attribute of the control. */
+  @property({ type: String })
+  public label!: string;
+
+  /** The placeholder attribute of the control. */
+  @property({ type: String })
+  public placeholder!: string;
+
+  /** Whether the dropdown's width should be the same as the target's one. */
+  @property({ type: Boolean, attribute: 'same-width' })
+  public override sameWidth = true;
+
+  /** The direction attribute of the control. */
+  @property({ reflect: true })
+  public override dir: 'ltr' | 'rtl' | 'auto' = 'auto';
+
+  constructor() {
+    super();
+    this.size = 'medium';
+  }
+
+  /** Sets focus on the component. */
+  public override focus(options?: FocusOptions) {
+    this.target.focus(options);
+  }
+
+  /** Removes focus from the component. */
+  public override blur() {
+    this.target.blur();
+  }
+
+  /** Checks the validity of the control. */
+  public reportValidity() {
+    this.invalid = this.required && !this.value;
+    if (this.invalid) this.focus();
+    return !this.invalid;
+  }
+
+  public override async firstUpdated() {
+    super.firstUpdated();
+    await this.updateComplete;
+
+    const selectedItem = this.allItems.find((i) => i.selected) ?? null;
+    if (selectedItem) this.value = selectedItem.value;
+    else this.updateSelected();
+  }
+
+  @watch('selectedItem')
+  protected updateValue() {
+    this.value = this.selectedItem?.value ?? this.value;
```

Comment:
This looks like it can lead to some odd states, if the value of the item is falsy somehow (I know, highly unlikely but still) and maintain the previous value. Basically, if there's a `selectedItem`, its value goes regardless what it is